### PR TITLE
js: add support for keydown events

### PR DESCRIPTION
**Pull request by leonklingele**
This will modify the player behavior even if the player element is unfocused.

Based on the YouTube key bindings, allow to

- toggle playback with space and 'k' key
- increase and decrease player volume with up / down arrow key
- mute and unmute player with 'm' key
- jump forwards and backwards by 5 seconds with right / left arrow key
- jump forwards and backwards by 10 seconds with 'l' / 'j'  key
- set video progress with number keys 0–9
- toggle captions with 'c' key
- toggle fullscreen mode with 'f' key
- play next video with 'N' key
- increase and decrease playback speed with '>' / '<' key

Also, remove a JavaScript dependency (`videojs.hotkeys.min.js`) which is no longer required.

### CODE REVIEW

**Comment by leonklingele** [comment]
Fixed an issue where changing the captions via the captions menu item would interfere with changing it through the 'c' keyboard shortcut.

File: assets/js/watch.js [hunk 1]
```diff
@@ -73,29 +73,33 @@ if (continue_button) {
     continue_button.onclick = continue_autoplay;
 }
 
-function continue_autoplay(event) {
```
Comment (leonklingele) [inline assets/js/watch.js]: Don't be fooled by the way GitHub shows this diff on the web UI.

This is a more accurate way of displaying it (`git -c diff.algorithm=patience log -U`):

```patch
diff --git a/assets/js/watch.js b/assets/js/watch.js
index 05e3b7e..0f3e812 100644
--- a/assets/js/watch.js
+++ b/assets/js/watch.js
@@ -75,2 +75,25 @@ if (continue_button) {

+function next_video() {
+    var url = new URL('https://example.com/watch?v=' + video_data.next_video);
+
+    if (video_data.params.autoplay || video_data.params.continue_autoplay) {
+        url.searchParams.set('autoplay', '1');
+    }
+
+    if (video_data.params.listen !== video_data.preferences.listen) {
+        url.searchParams.set('listen', video_data.params.listen);
+    }
+
+    if (video_data.params.speed !== video_data.preferences.speed) {
+        url.searchParams.set('speed', video_data.params.speed);
+    }
+
+    if (video_data.params.local !== video_data.preferences.local) {
+        url.searchParams.set('local', video_data.params.local);
+    }
+
+    url.searchParams.set('continue', '1');
+    location.assign(url.pathname + url.search);
+}
+
 function continue_autoplay(event) {
@@ -78,22 +101,3 @@ function continue_autoplay(event) {
         player.on('ended', function () {
-            var url = new URL('https://example.com/watch?v=' + video_data.next_video);
-
-            if (video_data.params.autoplay || video_data.params.continue_autoplay) {
-                url.searchParams.set('autoplay', '1');
-            }
-
-            if (video_data.params.listen !== video_data.preferences.listen) {
-                url.searchParams.set('listen', video_data.params.listen);
-            }
-
-            if (video_data.params.speed !== video_data.preferences.speed) {
-                url.searchParams.set('speed', video_data.params.speed);
-            }
-
-            if (video_data.params.local !== video_data.preferences.local) {
-                url.searchParams.set('local', video_data.params.local);
-            }
-
-            url.searchParams.set('continue', '1');
-            location.assign(url.pathname + url.search);
+            next_video();
         });

```

**Comment by omarroth** [comment]
Very nice, would appear to close #501 and #667.

Since this PR is adding support for hotkeys page-wide it should be possible to remove [L47-L101](https://github.com/omarroth/invidious/blob/master/assets/js/player.js#L47-L101) in player.js. 

For `Unhandled key down event`s, is it reasonable to ignore these or forward them to the page if the player doesn't handle them?

**Comment by omarroth** [comment]
Looks like you can also do:
```diff
diff --git a/assets/js/watch.js b/assets/js/watch.js
index 0f3e812..c8dd729 100644
--- a/assets/js/watch.js
+++ b/assets/js/watch.js
@@ -415,20 +415 @@ if (video_data.play_next) {
-        var url = new URL('https://example.com/watch?v=' + video_data.next_video);
-
-        if (video_data.params.autoplay || video_data.params.continue_autoplay) {
-            url.searchParams.set('autoplay', '1');
-        }
-
-        if (video_data.params.listen !== video_data.preferences.listen) {
-            url.searchParams.set('listen', video_data.params.listen);
-        }
-
-        if (video_data.params.speed !== video_data.preferences.speed) {
-            url.searchParams.set('speed', video_data.params.speed);
-        }
-
-        if (video_data.params.local !== video_data.preferences.local) {
-            url.searchParams.set('local', video_data.params.local);
-        }
-
-        url.searchParams.set('continue', '1');
-        location.assign(url.pathname + url.search);
+        next_video();
```

**Comment by leonklingele** [comment]
Applied your patch, changed a few `let`s to `const`s, updated function `increase_playback_rate` to rely on the playback rates available from the preferences, removed js dependency `videojs.hotkeys.min.js` which might interfere with out own hotkeys.

PTAL @omarroth

File: src/invidious/views/licenses.ecr [hunk 2]
```diff
@@ -136,17 +136,9 @@
         </tr>
 
         <tr>
-            <td>
-                <a href="/js/videojs.hotkeys.min.js?v=<%= ASSET_COMMIT %>">videojs.hotkeys.min.js</a>
-            </td>
-
             <td>
                 <a href="http://www.apache.org/licenses/LICENSE-2.0">Apache-2.0-only</a>
             </td>
```
Comment (omarroth) [inline src/invidious/views/licenses.ecr]: Shouldn't this also be removed?

File: assets/js/player.js [hunk 3]
```diff
@@ -38,69 +38,7 @@ var shareOptions = {
...
-                handler: function (player, _, e) {
-                    size = options.playbackRates.length;
-                    index = options.playbackRates.indexOf(player.playbackRate());
-                    player.playbackRate(options.playbackRates[(size + index - 1) % size]);
-                }
-            }
-        }
-    });
-});
+var player = videojs('player', options, function () { });
```
Comment (omarroth) [inline assets/js/player.js]: You can shorten this to:
```javascript
var player = videojs('player', options);
```

**Comment by omarroth** [comment]
I would still like hover scroll to be supported. You might take a look at [the relevant code videojs-hotkeys](https://github.com/ctd1500/videojs-hotkeys/blob/bb4a158b2e214ccab87c2e7b95f42bc45c6bfd87/videojs.hotkeys.js#L292-L328) for how it could be implemented.

Left a couple comments. Quite excited to merge this.

File: src/invidious/views/licenses.ecr (hunk 2 above)
Comment (leonklingele) [reply src/invidious/views/licenses.ecr]: Good catch, removed.

**Comment by leonklingele** [comment]
What's the benefit of having support for scroll-based volume-adjustments compared to the good old slider which can be adjusted by moving the mouse?

**Comment by omarroth** [comment]
It makes it possible to change volume without switching focus to another window. It's mostly a QoL feature, however since both Invidious and YouTube currently support it I would like for it to still be available.

**Comment by leonklingele** [comment]
Brought back support for volume-scrolling.

PTAL @omarroth

**Comment by leonklingele** [comment]
Rebased to latest `master`, CI should pass again.

**Comment by leonklingele** [comment]
CI is failing due to a bug in the Crystal build toolchain: https://travis-ci.org/omarroth/invidious/jobs/572082045#L279-L286
See also https://github.com/crystal-lang/crystal/issues/7558.

@omarroth could you please trigger a CI rebuild?

**Comment by omarroth** [comment]
LGTM :+1:
